compiler/rustc_mir_build/src/thir/constant.rs

```diff
@@ -102,12 +103,20 @@ pub(crate) fn parse_float<'tcx>(
                 Double::from_bits(rust_f.to_bits().into()),
                 rust_f.to_bits()
             );
+
             if neg {
                 f = -f;
             }
-            Scalar::from_f64(f)
+
+            Some(Scalar::from_f64(f))
         }
-    };
+    }
+}
 
-    Some(ConstValue::Scalar(scalar))
+fn parse_float_into_valtree<'tcx>(
```

Comment:
this function should not exist, a valtree of type float is a bug. Floating point numbers do not have structural equality so we should not support them in valtrees.

Instead either the `lit_to_const` impl to emit a `delay_span_bug` in that branch and return `Error(LitToConstError::Reported)` or even better, just remove that branch and return `LitToConstError::TypeError` here.